tests/pyspark/test_pyspark_decorators.py

```diff
@@ -0,0 +1,126 @@
+"""This module is to test the behaviour change based on defined config in pandera"""
+# pylint:disable=import-outside-toplevel,abstract-method
+
+from contextlib import nullcontext as does_not_raise
+import logging
+import pyspark.sql.types as T
+from pyspark.sql import DataFrame
+import pytest
+
+from pandera.backends.pyspark.decorators import cache_check_obj
+from pandera.config import CONFIG
+from pandera.pyspark import (
+    Check,
+    DataFrameSchema,
+    Column,
+)
+from tests.pyspark.conftest import spark_df
+
+
+class TestPanderaDecorators:
+    """Class to test all the different configs types"""
+
+    sample_data = [("Bread", 9), ("Cutter", 15)]
+
+    def test_pyspark_cache_requirements(self, spark, sample_spark_schema):
+        """Validates if decorator can only be applied in a proper function."""
+        # Set expected properties in Config object
+        CONFIG.pyspark_cache = True
+        input_df = spark_df(spark, self.sample_data, sample_spark_schema)
+
+        class FakeDataFrameSchemaBackend:
+            """Class that simulates DataFrameSchemaBackend class."""
+
+            @cache_check_obj()
+            def func_w_check_obj_args(self, check_obj: DataFrame, /):
+                """Right function to use this decorator, check_obj as arg."""
+                return check_obj.columns
+
+            @cache_check_obj()
+            def func_w_check_obj_kwargs(self, *, check_obj: DataFrame = None):
+                """Right function to use this decorator, check_obj as kwarg."""
+                return check_obj.columns
```

Comment:
`check_obj` can be passed as an `arg` or a `kwarg` now. Unit tests were added too.
Thank you for noting that @maxispeicher 👍